proto/process/agent.proto

```diff
@@ -97,12 +96,33 @@ message CollectorConnections {
 
 	map<string, RuntimeCompilationTelemetry> compilationTelemetryByAsset = 21;
 
+	repeated Route routes = 31;
+
+	repeated RouteMetadata routeMetadata = 34;
+
+	/////////////////
+	// V1 encodings
+
+	// encoded dns is a map of (string) ip-> list of domains (DNSEntry)
+	bytes encodedDNS = 14;
+
 	// All queried DNS domains for which we have collected stats
 	repeated string domains = 30;
 
-	repeated Route routes = 31;
+	//////////////////
+	// V2 encodings
+	// all domain strings are now indexes into a single array of strings
+	// hostname query information
+	bytes encodedDomainDatabase = 35;
```

Comment:
To make this code easier to consume, could we further explain the encoding format here  and for indexedDomains?  Or point to where the encoding format is defined.  Same comment about `encodedDNSLookups`.
  
For example:  are the strings encoded with a length tag? Or with null terminators?